### DISCUSSION TITLE
Guition JC8048W550 (C) display also works

### DISCUSSION THREAD

**Discussion by Acrobat77** [rzeldent/esp32-smartdisplay#185] [General]
Hi,

First a big thank you to rzeldent! First time user of LVGL here and everything (demo) worked on the first go without any changes. 

Of course I bought some displays before looking into the setup process and at first I thought I should have bought something else because of user reports I read regarding the lack of support for the ST7262. Then through Google I ended up here and -it just works-.

Good news: I bought a Guition JC8048W550 display (-C for capacitive touch) and looking at the example code I figured it was almost identical to the CYD 8048S050C. The Guition has a battery connector which I guess is a bonus and it is cheaper. 

Found it in the Guition Store here: https://a.aliexpress.com/_EGlCxAF
There’s a 4.3” version as well but that one seems to have less flash memory (4Mbyte).

PS: it’s not as blue as on the photo! :)

![IMG_2526](https://github.com/rzeldent/esp32-smartdisplay/assets/154474510/b291c2ef-7948-4cca-873a-c18816e86281)
![IMG_2527](https://github.com/rzeldent/esp32-smartdisplay/assets/154474510/c763dfe9-5607-46fe-95af-fecdbc423444)

**Comment by SoupySoups** [comment]
Can confirm, it works.

**Comment by mattjohnpowell** [comment]
I also can confirm. Made a little clock app as a tester. Nothing complicated but you can use it as a base

https://github.com/mattjohnpowell/clocky

**Reply by 82amp** [reply]
I tried to compile your code, and getting errors in esp32_smartdisplay/src/lvgl_panel_st7262_par.c

Looks like there might have been a change that has broken in the last 2 weeks?

disclaimer: cloning and runnig your repo has been my first platformio experience, I am completely green to it

Small snippet: 
...
PACKAGES: 
 - framework-arduinoespressif32 @ 3.0.7 
 - framework-arduinoespressif32-libs @ 5.1.0+sha.632e0c2a9f 
 - tool-esptoolpy @ 4.8.1 
 - tool-mklittlefs @ 3.2.0 
 - tool-riscv32-esp-elf-gdb @ 11.2.0+20220823 
 - tool-xtensa-esp-elf-gdb @ 11.2.0+20230208 
 - toolchain-riscv32-esp @ 12.2.0+20230208
 - toolchain-xtensa-esp32s3 @ 12.2.0+20230208
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 42 compatible libraries
Scanning dependencies...
Dependency Graph
|-- esp32_smartdisplay @ 2.1.0+sha.5e0221f
|-- WiFi @ 3.0.7
|-- lvgl @ 9.2.2
...
.pio/libdeps/esp32-8048S050C/esp32_smartdisplay/src/lvgl_panel_st7262_par.c: In function 'lvgl_lcd_init':
.pio/libdeps/esp32-8048S050C/esp32_smartdisplay/src/lvgl_panel_st7262_par.c:101:10: error: 'esp_lcd_rgb_panel_config_t' has no member named 'on_frame_trans_done'
  101 |         .on_frame_trans_done = direct_io_frame_trans_done,

**Reply by 82amp** [reply]
Okay, so when PlatformIO was installed, for some reason, it pulled an old version of Espressif 32. - it was @51.3 or something. I've manually installed 6.9.0, and uninstalled 51.3, and it is now compiling!

**Comment by Adolfovik** [comment]
Hello, I am trying to develop my own project with this board (JC4840W550), and I can compile and run with Platformio both examples of rzeldent and mattjohnpowell clock (very nice). But, in both cases, I have an annoying jitter effect in the display associated (I think so) with screen refresh.
Do not have you this jitter effect?

**Reply by SoupySoups** [reply]
Adjust pclk

**Reply by Adolfovik** [reply]
Thanks, but I got it finally without LVGL and without Platformio. I was not familiar with those tools, so I did it only with Arduino and GFX library. I will publish soon here.

**Reply by Adolfovik** [reply]
Finally here is my weather station using the JC4840W550 board with GFX library:

https://github.com/Adolfovik/MultiMeteo

I hope can help anyone fighting with this cheap but laborious board.

![IMG_20250525_175227_edit_3520816798345567](https://github.com/user-attachments/assets/0dfcd886-3d3f-4798-a3bc-b1907cd95d63)
![IMG_20250601_085347](https://github.com/user-attachments/assets/a4f15d28-427f-41c4-a4da-0c5e5b7746c4)

**Comment by vijugo** [comment]
Good morning. I had a piece of software that worked very well with Squareline and an ESP32S3 on a JC8048W550 display. About a month ago, I received some new models with a different ESP32S3 reference. When I loaded the program, the image remained blank. The ESP ran normally, and the display worked, but I didn't see any images. Could someone help me figure out how to get the images to appear? I've attached the software in case anyone could help me. I'd really appreciate it. https://drive.google.com/file/d/1SS6X_I-WKKLQWw33BCKnvBBp-eKFN2cX/view?usp=sharing

**Reply by Ebusicano** [reply]
has conseguido arreglarlo? Podrias pasarme el proyecto de la JC8048W550? M estoy volviendo loco con esta pantalla

**Reply by vijugo** [reply]
Ebusicano, sinceramente yo le trabaje 5 días tratando de que me funcionara, pero no lo logre, como tenía que entregar un proyecto use una plataforma  china similar a Square Line , pero con menos parámetros y hay si funciona la pantalla. Pero el inconveniente que tiene es que el esp32 trabaja independiente y no puedo acceder como lo puedo con la libreria smartdisplay. Se requiere de un microcontrolador externo y se controla a través de comando json. Por ahora estoy usando esto mientras que alguien de una solución. https://www.guition.com/main/

**Reply by Ebusicano** [reply]
@vijugo Por fin la hice funcionar, tras dias y dias investigando he conseguido integrar una interfaz con eez flow. Lo malo, que tengo un flickering brutal tipo efecto glitch y no se como solucionarlo. Supongo que sea cosa de los drivers, pero de momento no ayuda ni ChatGPT.

**Reply by vijugo** [reply]
Ebusicano, si me puede pasar un programa de prueba  básico yo logre mejorar eso con un timer y lo trabajo para ver como lo puedo mejorar.

**Comment by vijugo** [comment]
// Configuración del temporizador
void setup_lvgl_timer() {
    timer_LGVL = timerBegin(1, 80, true);  // Temporizador 0, prescaler 80 (1us por tick)
    timerAttachInterrupt(timer_LGVL, &lv_tick_handler, true);
    timerAlarmWrite(timer_LGVL, 1000, true);  // Interrupción cada 1000us = 1ms
    timerAlarmEnable(timer_LGVL);
}

void InicializarPantalla(void){
    // Initialize the display
    smartdisplay_init();
    auto disp = lv_disp_get_default();
    setup_lvgl_timer();  // Initialize the timer to update LVGL 
    lv_disp_set_rotation(disp, LV_DISP_ROTATION_0);
     ui_init();
    // Initialize the screens
}

**Reply by Ebusicano** [reply]
@vijugo el archivo del codigo pesa demasiado, alguna otra forma de enviarte el proyecto?

**Reply by Ebusicano** [reply]
@vijugo Arreglado, ajustando el pclk de 16 a 15.4

**Comment by vijugo** [comment]
@Ebusicano  muchas gracias, efectivamente el cambio que hay que hacer es en la carpeta boards ubica el archivo json que esta utilizando para mi caso esp32-8048S050C y donde aparece  "'-D ST7262_PANEL_CONFIG_TIMINGS_PCLK_HZ=(12.5*1000000)'", lo modifico a  "'-D ST7262_PANEL_CONFIG_TIMINGS_PCLK_HZ=(15.4*1000000)'" y la pantalla funciona. Si alguien tiene dudas con respecto al dispostivo que esta utilizando por favor  abran el archivo platformio.ini ubico la linea default_envs = esp32-8048S050C. 
@Ebusicano  mil gracias por su colaboración y estoy dispuesto a colaborarle en lo que necesite, si esta a mi alcance.

**Comment by strange-v** [comment]
ESPHome now supports this display via the new `mipi_rgb` platform.

But I have a question for you folks: do your displays also show a slight greenish tint / incorrect white balance? When I show two identical dashboards on this screen and the Guition ESP32-S3-4848S040, it’s clearly visible that the latter has much better colors, matching what I see on a monitor.

**Comment by crossseasub-svg** [comment]
hello everyone. i have the same model jc8048w550C_I everything works except that the screen is a little shaking i tried every thing but in vain. any help please

**Reply by DaverDavids** [reply]
> hello everyone. i have the same model jc8048w550C_I everything works except that the screen is a little shaking i tried every thing but in vain. any help please

Did you adjust PCLK to 15.4MHz? Also I found changing bounce_buffer_size to a multiple of 800 helped smooth things. Though I am still debugging some things on this display
